packages/rum-core/src/domain/view/viewMetrics/trackCumulativeLayoutShift.ts

```diff
@@ -82,26 +82,39 @@ export function trackCumulativeLayoutShift(
   }
 }
 
-function slidingSessionWindow() {
+const MAX_WINDOW_LENGTH = 5 * ONE_SECOND
+const MAX_UPDATE_GAP = ONE_SECOND
+
+export function slidingSessionWindow() {
   let value = 0
   let startTime: RelativeTime
   let endTime: RelativeTime
 
   let largestLayoutShift = 0
   let largestLayoutShiftTarget: HTMLElement | undefined
   let largestLayoutShiftTime: RelativeTime
+  let timeoutId: TimeoutId
+
+  function resetWindow(entry: RumLayoutShiftTiming) {
+    startTime = endTime = entry.startTime
+    value = entry.value
+    largestLayoutShift = 0
+    largestLayoutShiftTarget = undefined
+    largestLayoutShiftTime = entry.startTime
+  }
+
   return {
     update: (entry: RumLayoutShiftTiming) => {
+      clearTimeout(timeoutId)
+      timeoutId = setTimeout(() => resetWindow(entry), MAX_WINDOW_LENGTH)
```

Comment:
**💬 suggestion:** ‏Since a window cannot be longer than `MAX_WINDOW_LENGTH`, I think it doesn't make sense to set the timeout like that:
* keep the timeout here (reseting at every new entry) and use a timeout duration of `MAX_UPDATE_GAP`
* or move the timeout when creating the new window and keep using `MAX_WINDOW_LENGTH` (I think that would me more efficient)

**💭 thought:** ‏I don't think it makes sense to call `resetWindow` with the `entry` here. `resetWindow` should reset the state without using any entry, since it is to clear up the current window.

Reply:
**💭 thought:** ‏Since the PerformanceObserver is reporting timings asynchronously, `entry.startTime` might be in the past. Resetting the window after `MAX_WINDOW_LENGTH` might be too early and we might miss some entries that would normally it in the window. Maybe we can wait a little be longer just to be safe